Src/PerformanceCollector/Shared/Implementation/StandardPerformanceCollector/CounterFactory.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.StandardPerformanceCollector
+{
+    using System;
+
+    /// <summary>
+    /// Factory to create different counters.
+    /// </summary>
+    internal class CounterFactory
+    {
+        /// <summary>
+        /// Gets a counter.
+        /// </summary>
+        /// <param name="originalString">Original string definition of the counter.</param>
+        /// <param name="categoryName">Category name.</param>
+        /// <param name="counterName">Counter name.</param>
+        /// <param name="instanceName">Instance name.</param>
+        /// <returns>The counter identified by counter name.</returns>
+        internal ICounterValue GetCounter(string originalString, string categoryName, string counterName, string instanceName)
```

Comment:
Can it be aligned with the factory for azure web site? Move parsing of `originalString` here